extension/storage/filestorage/client.go

```diff
@@ -50,44 +50,86 @@ func newClient(filePath string, timeout time.Duration) (*fileStorageClient, erro
 }
 
 // Get will retrieve data from storage that corresponds to the specified key
-func (c *fileStorageClient) Get(_ context.Context, key string) ([]byte, error) {
-	var result []byte
+func (c *fileStorageClient) Get(ctx context.Context, key string) ([]byte, error) {
+	results, err := c.GetBatch(ctx, []string{key})
+	if err != nil {
+		return nil, err
+	}
+	return results[0], nil
+}
+
+// GetBatch will retrieve data from storage that corresponds to the specified collection of keys in a single transaction
+func (c *fileStorageClient) GetBatch(_ context.Context, keys []string) ([][]byte, error) {
+	results := make([][]byte, len(keys))
```

Comment:
Specify the capacity as we know how many results are gonna be there right? 

```suggestion
	results := make([][]byte, len(keys), len(keys))
```

Reply:
I take that back actually, not needed when cap is expected to be == len.